packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -261,6 +261,21 @@ export class ApolloServerBase<
         : noIntro;
     }
 
+    if (
+      this.config.nodeEnv === 'production' &&
+      !requestOptions.cache &&
+      (requestOptions.persistedQueries === undefined ||
+        (requestOptions.persistedQueries &&
+          !requestOptions.persistedQueries.cache &&
+          !requestOptions.persistedQueries.ttl))
+    ) {
+      this.logger.warn(
+        'Apollo Server is running with an unbounded in-memory cache in production. ' +
```

Comment:
Persisted queries are enabled and use an unbounded cache. Your server is vulnerable to denial of service attacks via memory exhaustion. Set `cache: 'bounded'` or `persistedQueries: false` in your ApolloServer constructor, or see DOCS for other alternatives.